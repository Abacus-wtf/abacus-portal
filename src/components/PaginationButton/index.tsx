import React, { FunctionComponent } from "react"
import Button from "../Button"

type PaginationButtonProps = {
  isLastPage: boolean
  isLoading: boolean
  getNextPage: () => void
}

const PaginationButton: FunctionComponent<PaginationButtonProps> = ({
  isLastPage,
  isLoading,
  getNextPage,
}) => {
  if (isLastPage || isLoading) {
    return null
  }

  const handleClick = () => {
    getNextPage()
  }

  return (
    <Button type="button" onClick={handleClick} disabled={isLoading}>
      Show More
    </Button>
  )
}

export default PaginationButton
