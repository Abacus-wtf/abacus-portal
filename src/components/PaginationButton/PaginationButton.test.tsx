import { render } from "@test-utils/index"
import React from "react"
import "@testing-library/jest-dom/extend-expect"
import PaginationButton from "."

const defaultProps = {
  isLastPage: false,
  isLoading: false,
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  getNextPage: () => {},
}

describe("Pagination Button", () => {
  it("Matches snapshot", () => {
    const component = render(<PaginationButton {...defaultProps} />)
    expect(component.container).toMatchSnapshot()
  })

  it("Should be null when loading", () => {
    const { container } = render(
      <PaginationButton {...defaultProps} isLoading />
    )
    expect(container.firstChild).toBeNull()
  })

  it("Should be null when last page", () => {
    const { container } = render(
      <PaginationButton {...defaultProps} isLastPage />
    )
    expect(container.firstChild).toBeNull()
  })
})
