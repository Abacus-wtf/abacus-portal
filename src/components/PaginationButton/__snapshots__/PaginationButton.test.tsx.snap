// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Button Matches snapshot 1`] = `
<div>
  <button
    class="sc-bdvvtL gAgHBR btn btn-primary"
    type="button"
  >
    Show More
  </button>
</div>
`;
